### ISSUE TITLE
"unknown tool name found in scoped lint: `rustdoc::all`" on older Rust versions

### ISSUE THREAD

**Issue by Spaqin** [phip1611/tar-no-std#11]
Hi, while working on a feature for our open-source control system for quantum physics experiments - [ARTIQ](https://github.com/m-labs/artiq/) - I found this crate to be potentially very useful.

However, trying to integrate it, I did encounter a small issue:

```
error[E0710]: an unknown tool name found in scoped lint: `rustdoc::all`
  --> /home/spaqin/.cargo/registry/src/github.com-1ecc6299db9ec823/tar-no-std-0.1.8/src/lib.rs:77:9
   |
77 | #![deny(rustdoc::all)]
   |         ^^^^^^^

error: aborting due to previous error

For more information about this error, try `rustc --explain E0710`.
error: could not compile `tar-no-std`
```

First things first: the problem is that we're using a nightly Rust version from 2021-01-29, rather than stable 1.52+ that includes ``rustdoc``. With complex projects updating the toolchain should be done with utmost care, as things may (and do) break unexpectedly. Fixing that takes a lot of time, so we're not willing to update unless really necessary.

On the other hand, just removing that one line, allows it to work without an issue: https://github.com/spaqin/tar-no-std/commit/2ab6dc58e5249c59c4eb03eaf3a119bcdd678d32

I realize that removing it from the main repository could potentially break your documentation tests, so my suggestion for a compromise would be to do it only for 0.1.x line for legacy support (making it actually compatible with ``2018`` edition as specified in your ``Cargo.toml`` for the older version), and keep your current main as-is.

**Comment by phip1611** [comment]
Hi @Spaqin ,
thanks for the interest in my library. Sounds like a cool project you're working on!

TL;DR: Please fork this and reference it as git dependency. This might be the quickest solution for you.

> With complex projects updating the toolchain should be done with utmost care, as things may (and do) break unexpectedly.

While I generally agree with you, especially for complex, C/C++ toolchains, I don't think that this statement holds for Rust toolchains. Especially because of Rust's _stability without stagnation_ statement, the Rust editions etc.

IMHO, I recommend you to spend time on improving the infrastructure of your repository so that toolchain updates are an easy and quickly done with sufficient CI coverage.

Also the general Rust ecosystem tries to stay up-to-date with new Rust releases. 
If its not my library, you will face the same problem with the next one.

Last but not least, having fresh software and fresh toolchains is one of my personal important properties in software development. So it feels a bit unnatural for me to do such a change to support such an old toolchain.


Sorry if I disappoint you. Please fork this repository and reference it as git dependency. This might be the quickest solution for you.
